Mathlib/Algebra/Module/Zlattice.lean

```diff
@@ -153,6 +162,15 @@ theorem fract_mem_fundamentalDomain (x : E) : fract b x ∈ fundamentalDomain b
   fract_eq_self.mp (fract_fract b _)
 #align zspan.fract_mem_fundamental_domain Zspan.fract_mem_fundamentalDomain
 
+/-- The map `fract` with codomain restricted to `fundamentalDomain`. -/
+def fract_restrict (x : E) : fundamentalDomain b := ⟨fract b x, fract_mem_fundamentalDomain b x⟩
```

Comment:
I think it would be better to just define `fract` as this bundled version; what do you think?

Reply:
I am not sure. In  almost all results, I will then need to coerce `fract` to an element of `E` to state the result (except one or two). Furthermore, since `fundamentalDomain` has no algebraic structure, I will need also to coerce `fract` to an element of `E` to do any significant computation. 

I can try if you want. What would be the expected gain though?